### PR TITLE
1.11.1 cherry pick

### PR DESCRIPTION
**Pull request by mlaventure**
cherry-picks for 1.11.1-rc1

PRs Included:
#22318 61c0f4b Update runc and spec dependencies for mount label
#22125 ab1be6f Reset restart timeout if execution longer than 10s
#22254 c06a4c2 Add default `serveraddress` value in remote API `/auth`
#22299 3b8c11b fix reloading of daemon labels from config
#22184 0c8c9ba Labels set on the command line always override labels in Dockerfile
#22256 dc98419 Use absolute path for rootfs in OCI config.json
#22255 f503494 Bump containerd to v0.2.1
#22253 91880b4 api: server: check for unauthorized error
#22237 75d6341 Fix config cleanup on canceling restartmanager
#22015 2735744 add pkcs11 build tag to deb and rpm builds
#22213 45730a4 Docker http panics caused by container deletion with empty names.
#22172 ce9e2d8 Containerize `TestDaemonNoSpaceLeftOnDeviceError`
#22191 c3348d0 Initialize activateWait for plugins activated by json spec
#22123 8c1d802 Remove restart canceled error
#22121 e04dca2 Clean up exec fifos on process exit
#22120 9d68491 Create a copy of stats value before modifications
#22065 ccfb894 Remove deprecation warning
#21993 c941748 Don't throw "restartmanager canceled" error for no restart policy container
#22058 044398a Remove rpc error when shut down daemon
#22048 41130fb docs: update API for features added in 1.11
#21949 9d99e36 Vendor distribution to correct config blob media type in schema2 manifest

This also fixes the content of `VERSION` to be `1.11.1`

### CODE REVIEW

**Comment by mlaventure** [comment]
Push what should be the last version. 

PTAL and check that I haven't forgotten anything :)

File: daemon/container_operations_unix.go [hunk 1]
```diff
@@ -311,7 +311,7 @@ func getDevicesFromPath(deviceMapping containertypes.DeviceMapping) (devs []spec
 
 	// check if it is a symbolic link
 	if src, e := os.Lstat(deviceMapping.PathOnHost); e == nil && src.Mode()&os.ModeSymlink == os.ModeSymlink {
-		if linkedPathOnHost, e := os.Readlink(deviceMapping.PathOnHost); e == nil {
+		if linkedPathOnHost, e := filepath.EvalSymlinks(deviceMapping.PathOnHost); e == nil {
```
Comment (icecrime) [inline daemon/container_operations_unix.go]: This is fixing a P3, it has nothing to do in a patch release.

**Comment by mlaventure** [comment]
win2lin failed due to a network issue:
```
01:56:36 ----------------------------------------------------------------------
01:56:36 FAIL: docker_cli_build_test.go:2911: DockerSuite.TestBuildAddRemoteFileMTime
01:56:36 
01:56:36 docker_cli_build_test.go:2966:
01:56:36     c.Fatal(err)
01:56:36 ... Error: failed to build the image: Sending build context to Docker daemon 2.048 kB

01:56:36 Step 1 : FROM scratch
01:56:36  ---> 
01:56:36 Step 2 : MAINTAINER dockerio
01:56:36  ---> Using cache
01:56:36  ---> 4e15856e030a
01:56:36 Step 3 : ADD http://10.0.0.12:32771/baz /usr/lib/baz/quux
01:56:36 Get http://10.0.0.12:32771/baz: dial tcp 10.0.0.12:32771: getsockopt: connection refused
01:56:36 
01:56:36 
01:56:39 
01:56:39 ----------------------------------------------------------------------
```

I'd say good to merge :)

**Comment by tonistiigi** [comment]
LGTM

**Comment by tiborvass** [comment]
LGTM

**Pull request merged**